kasa/tests/fakeprotocol_smartcamera.py

```diff
@@ -0,0 +1,213 @@
+from kasa.experimental.smartcameraprotocol import SmartCameraProtocol
+from kasa.protocol import BaseTransport
+from kasa.smart import SmartChildDevice
+
+
+class FakeSmartCameraProtocol(SmartCameraProtocol):
+    def __init__(self, info, fixture_name):
+        super().__init__(
+            transport=FakeSmartCameraTransport(info, fixture_name),
+        )
+
+    async def query(self, request, retry_count: int = 3):
+        """Implement query here so can still patch SmartProtocol.query."""
+        resp_dict = await self._query(request, retry_count)
+        return resp_dict
+
+
+class FakeSmartCameraTransport(BaseTransport):
+    def __init__(
+        self,
+        info,
+        fixture_name,
+        *,
+        list_return_size=10,
+    ):
+        super().__init__(
+            config=DeviceConfig(
+                "127.0.0.123",
+                credentials=Credentials(
+                    username="dummy_user",
+                    password="dummy_password",  # noqa: S106
+                ),
+            ),
+        )
+        self.fixture_name = fixture_name
+        self.info = copy.deepcopy(info)
+        self.child_protocols = self._get_child_protocols()
+        self.list_return_size = list_return_size
+
+    @property
+    def default_port(self):
+        """Default port for the transport."""
+        return 443
+
+    @property
+    def credentials_hash(self):
+        """The hashed credentials used by the transport."""
+        return self._credentials.username + self._credentials.password + "camerahash"
+
+    async def send(self, request: str):
+        request_dict = json_loads(request)
+        method = request_dict["method"]
+
+        if method == "multipleRequest":
+            params = request_dict["params"]
+            responses = []
+            for request in params["requests"]:
+                response = await self._send_request(request)  # type: ignore[arg-type]
+                # Devices do not continue after error
+                if response["error_code"] != 0:
+                    break
+                response["method"] = request["method"]  # type: ignore[index]
+                responses.append(response)
+            return {"result": {"responses": responses}, "error_code": 0}
+        else:
+            return await self._send_request(request_dict)
+
+    def _get_child_protocols(self):
+        child_infos = self.info.get("getChildDeviceList", {}).get(
+            "child_device_list", []
+        )
+        found_child_fixture_infos = []
+        child_protocols = {}
+        for child_info in child_infos:
+            if (
+                (device_id := child_info.get("device_id"))
+                and (category := child_info.get("category"))
+                and category in SmartChildDevice.CHILD_DEVICE_TYPE_MAP
+            ):
+                from .conftest import filter_fixtures
+                from .fakeprotocol_smart import FakeSmartProtocol
+
```

Comment:
Import on top of the file?

Reply:
The filter fixtures needs to be here to avoid circular imports but I've moved the other one to the top.